src/components/TextEntityRecognition/index.js

```diff
@@ -47,9 +48,22 @@ const entitySequenceToSimpleSeq = (doc, entSeq) => {
 
 export const TextEntityRecognition = (props) => {
   const [currentSampleIndex, changeCurrentSampleIndex] = useState(0)
+  const [textToShow, changeTextToShow] = useState("")
+  const oldText = useRef()
+  useEffect(() => {
+    if (oldText.current !== textToShow) {
+      datasetHelper
+        .getTextfromSample(props.samples[currentSampleIndex])
+        .then((result) => {
+          changeTextToShow(result)
+        })
+      oldText.current = textToShow
+    }
+  }, [props.samples, currentSampleIndex, textToShow])
```

Comment:
since this code is used multiple times, you could create a hook called `useSampleText` to simplify both interfaces

Reply:
I store it where in utils ?

Reply:
You could put it in utils at the top level